tools/freeze_requirements.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from concurrent.futures import ThreadPoolExecutor
+from os.path import abspath
+from subprocess import CalledProcessError, run
+
+from tools.lib import gitroot
+
+
+def main() -> int:
+    base_path = abspath(gitroot())
+    base_cmd = (
+        "pip-compile",
```

Comment:
Could you please add `pip-tools` to requirements-dev.txt?

Reply:
we'll eventually put `pip-tools` into the distributable dev bundle, but reqs-dev is a fine place for it for now!